### ISSUE TITLE
VarDictJava 1.5.5: negative values for fisher.test

### ISSUE THREAD

**Issue by chapmanb** [AstraZeneca-NGS/VarDictJava#144]
Polina and all;
We identified an additional negative value output when testing with VarDictJava 1.5.5. This comes from RNA-seq variant calling, so might be exposing some additional edge cases we didn't see in previous reports. When using a fairly vanilla command line:
```
vardict-java -G hg38.fa -f 0.1 -N sample -b in.bam  -c 1 -S 2 -E 3 -g 4  regions31_04.bed > vardict-out.tsv
cat vardict-out.tsv | teststrandbias.R
```
It exits out with:
```
Error in fisher.test(matrix(c(d[i, 10], d[i, 11], d[i, 12], d[i, 13]),  :
  all entries of 'x' must be nonnegative and finite
Execution halted
```
I'll send a reproducible example via e-mail to help diagnose. Please let me know if I can provide any other details at all.

**Comment by PolinaBevad** [comment]
@chapmanb , hello!
Thank you for reporting this issue!

It is fixed in #148, we adjust negative counts to zero and log the warning. Also, the warning is described here: https://github.com/AstraZeneca-NGS/VarDictJava/wiki/Warning-%22...-column-in-variant-on-position-was-negative,-adjusted-to-zero%22  

I will implement the better solution with checking introns suggested by Zhongwu in one of the next PRs. 
Test case was added to our set. Thank you for its preparation!

**Comment by roryk** [comment]
Thanks so much everyone.

**Comment by PolinaBevad** [comment]
Hello all! 
I'll close this issue because PR #150 (with the check for indels before/after introns) merged and the release 1.5.7 contains this fix.
Thank you a lot for helping with test cases!
